tests/spin/test_lattice.py

```diff
@@ -539,3 +763,180 @@ def test_shape_error():
     lattice = "Octagon"
     with pytest.raises(ValueError, match="Lattice shape, 'Octagon' is not supported."):
         _generate_lattice(lattice=lattice, n_cells=n_cells)
+
+
+@pytest.mark.parametrize(
+    ("shape", "n_cells", "expected_number"),
+    # expected_number here was obtained manually.
+    [
+        ("fcc", [2, 2, 2], 32),
+        ("bcc", [2, 2, 2], 16),
+        ("kagome", [2, 2], 12),
+        ("lieb", [3, 3], 27),
+    ],
+)
+def test_num_sites_lattice_templates(shape, n_cells, expected_number):
+    r"""Test that the correct number of lattice points are generated for the given attributes"""
+    lattice = _generate_lattice(lattice=shape, n_cells=n_cells)
+    assert lattice.n_sites == expected_number
+
+
+@pytest.mark.parametrize(
+    # expected_points here were calculated manually.
+    ("shape", "n_cells", "expected_points"),
```

Comment:
Do we have such tests for all lattices or just for 3D ones?

Reply:
yes, we test the rest of the cases when we test the lattice class.